rosidl_runtime_cpp/test/test_bounded_vector.cpp

```diff
@@ -50,3 +52,39 @@ TEST(rosidl_generator_cpp, bounded_vector_rvalue) {
   ASSERT_EQ(vv[0], 1);
   ASSERT_EQ(vv[1], 2);
 }
+
+TEST(rosidl_generator_cpp, bounded_vector_comparisons) {
+  rosidl_runtime_cpp::BoundedVector<int, 2> v;
+  ASSERT_EQ(v, v);
+  const decltype(v) vv;
+  v.push_back(1);
+  ASSERT_NE(v, vv);
+  ASSERT_GT(v, vv);
+  ASSERT_LT(vv, v);
+}
+
+TEST(rosidl_generator_cpp, bounded_vector_input_iterators) {
+  rosidl_runtime_cpp::BoundedVector<int, 4> v;
+  std::istringstream ss("1 2 3");
+  std::istream_iterator<int> ii(ss), end;
+  v.assign(ii, end);
+  ASSERT_EQ(v.size(), 3);
```

Comment:
@jwakely checking `v`'s content would also be good. Same elsewhere.

Reply:
Done